### PR TITLE
Support ghc-8.10.

### LINKED ISSUE #65: Support GHC 8.10
I'd like to get some help with supporting this new release. Some items that likely require attention:

  * Getting a working build plan
  * Adapting to GHC internal API changes (if any)

### PR DESCRIPTION
**Pull request by judah**
The biggest API change is that ghc-8.10 organizes its modules into
a hierarchy, such as `GHC.Hs.Expr` instead of `HsExpr`. To work
around that without adding even more CPP, I added some `other-modules`
providing the new names for older GHC releases.

Fixes #65.

### CODE REVIEW

File: src/GHC/SourceGen/Syntax/Internal.hs [hunk 1]
```diff
@@ -103,7 +114,9 @@ builtLoc = L builtSpan
 -- In GHC-8.8, source locations for Pat aren't stored in each node, and
 -- LPat is a synonym for Pat.
```
Comment (jinwoo) [inline src/GHC/SourceGen/Syntax/Internal.hs:115]: Update this comment on 8.10?

Comment (judah) [reply src/GHC/SourceGen/Syntax/Internal.hs]: Updated comment and simplified the CPP to a single if/else. (The extra logic is *only* needed for 8.8, not for 8.4, 8.6 or 8.10.)

**Comment by judah** [comment]
I fixed the build failure; once of the wrapper modules didn't make it into the commit.

**Comment by jinwoo** [comment]
@judah I think this should be released to Hackage before other packages dependent on ghc-source-gen can support 8.10. Can you? Or do you want me to?

**Comment by judah** [comment]
@jinwoo sure, you can go ahead and release it. Do you have an account on Hackage? I can add you to the maintainers list.

**Comment by jinwoo** [comment]
Thank you, @judah . I just created an account, JinwooLee.

**Comment by judah** [comment]
OK, I've added you as a maintainer.

The `stack upload` command ought to do the job, but let me know if you get stuck.

**Comment by jinwoo** [comment]
Thank you, Judah!

**Comment by jinwoo** [comment]
When I do `stack upload .`, it fails with

```
You are not an authorized package uploader. Please contact the server trustees at hackage-trustees@haskell.org to request to be added to the Uploaders group.
```

I emailed hackage-trustees@haskell.org as instructed.

**Comment by jinwoo** [comment]
0.4.0.0 is now published.

**Pull request merged**